database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Views/PostCell.swift

```diff
@@ -30,6 +38,7 @@ struct PostCellView: View {
             Image(systemName: isStarred ? "star.fill" : "star")
               .onTapGesture {
                 isStarred.toggle()
+                post.didTapStarButton(isStarred: isStarred)
```

Comment:
nit: Seems like we don't follow MVVM pattern here since Post is a clear Model and View should never modify Model directly but only via ViewModel. The intention of such indirection is needed to decouple data layer from the UI related logic which is important for creating maintainable applications. It is probably not a huge issue for our tiny sample app and such direct approach may be even an advantage for demo purposes, but I'm a little bit worried that we are showing a bad example.

If we decide to modify it to canonical MVVM I can help with options.

@paulb777 @ryanwilson @ncooke3 Do you have an opinion on if we should keep following canonical MVVM or should compromise on it for more direct code?

Reply:
My opinion is that we should strive to demonstrate best practices with our Quick Starts.

Reply:
The most straightforward way I can see so far would be the following:
1. Introduce a reference type (class) like `PostViewModel`. It will have all required `@Published` properties to display the post cell and the methods for actions like the star button press.
2. Replace `UserViewModel.posts` type by `[PostViewModel]`
3. Initialize `PostCell` with a selected `PostViewModel`. The `PostViewModel` will be a `@State` property then and will be the single source of truth

Using bindings will not be required then because `PostViewModel` is a reference type. In this case `PostViewModel` can depend on anything required like with Post, userID, Firebase instance, etc. and will provide a binding point between UI and model layers.

Reply:
Thank you for the suggestion I think creating a `PostViewModel` is a good idea. I will make that change on the following commits.